hw-webapi/flask_app.py

```diff
@@ -0,0 +1,298 @@
+    /picture/<id>       PATCH   - update image url in DB
+    /picture/<id>       DELETE  - delete image url in DB
+"""
+
+import sqlite3
+from functools import wraps
+
+import requests
+from werkzeug.exceptions import abort
+
+from config import DB, DEFAULT_IMAGE, DEFAULT_QUOTE
+from flask import Flask, jsonify, request, url_for
+from waitress import serve
+
+app = Flask(__name__)
+
+
+def db_connect(func):
+    """Decorator to launch function execution on database"""
+
+    @wraps(func)
+    def wrapper_func(*a, **kw):
+        with sqlite3.connect(DB) as db:
+            crs = db.cursor()
+            ret_value = func(crs, *a, **kw)
+            db.commit()
+        return ret_value
+
+    return wrapper_func
+
+
+@db_connect
+def _get_random_entry(crs: sqlite3.Cursor, table: str) -> str:
+    """Get random entry from given table
+
+    Args:
+      crs: sqlite3.Cursor to execute sql query
+
+      table: table name to select random entry from
+
+    Returns:
+      Text value of entry
+    """
+    query_res = crs.execute(f"select * from {table} order by random() limit 1;")
+    if query_res:
+        query_res = query_res.fetchall()
+        return query_res[0][1] if query_res else str()
+    else:
+        return str()
+
+
+@db_connect
+def _get_entry_by_id(crs: sqlite3.Cursor, eid: int, table: str) -> list:
+    """Get specific entry from table by its id
+
+    Args:
+      crs: sqlite3.Cursor to execute sql query
+
+      eid: entry id
+
+      table: table name to select from
+
+    Returns:
+      A list of entries matching condition
+    """
+    query_res = crs.execute(f"select * from {table} where id=:eid", {"eid": eid})
+    if query_res:
+        return query_res.fetchall()
+    else:
+        return list()
+
+
+@db_connect
+def _del_entry_by_id(crs: sqlite3.Cursor, eid: int, table: str) -> int:
+    """Delete entry from table
+
+    Args:
+      crs: sqlite3.Cursor to execute sql query
+
+      eid: entry id
+
+      table: table name to delete from
+
+    Returns:
+      Integer ID of deleted entry
+    """
+    if crs.execute(f"delete from {table} where id=:eid", {"eid": eid}):
+        return eid
+    else:
+        return -1
+
+
+@db_connect
+def _upd_entry_by_id(crs: sqlite3.Cursor, eid: int, new_value: str, table: str) -> int:
+    """Update value of specific entry by its id
+
+    Args:
+      crs: sqlite3.Cursor to execute sql query
+
+      eid: entry id
+
+      new_value: value to set up as new one
+
+      table: table name for update operation
+
+    Returns:
+      Integer ID of updated entry
+    """
+    if crs.execute(
+        f"update {table} set txt=:new_val where id=:eid",
+        {"eid": eid, "new_val": new_value},
+    ):
+        return eid
+    else:
+        return -1
+
+
+@app.route("/", methods=["GET"])
+@app.route("/index", methods=["GET"])
+def index():
+    """Main page"""
+    return "<h1>Inspiring thoughts... Go to /next</h1>"
+
+
+@app.route("/next", methods=["GET"])
+def next_thought():
```

Comment:
Sometimes this view return error with code 500, could you catch exceptions and return a readable error in this case?
It also right for all endpoints, because errors sometimes it happens